indexing-service/src/main/java/org/apache/druid/indexing/common/TaskStorageDirTracker.java

```diff
@@ -154,6 +159,7 @@ public synchronized StorageSlot pickStorageSlot(String taskId)
   public synchronized void returnStorageSlot(StorageSlot slot)
   {
     if (slot.getParentRef() == this) {
+      --numUsedSlots;
```

Comment:
Please do this after the `runningTaskId` has been set to null.
It wouldn't make any functional difference but makes for better symmetry.

The order should be:
increment slot count -> use slot -> give up slot -> decrement slot count.